mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -83,9 +86,12 @@ class NaiveBayesModel private[spark] (
   }
 
   override def predict(testData: RDD[Vector]): RDD[Double] = {
-    val bcModel = testData.context.broadcast(this)
+    bcModel match {
+      case None => bcModel = Some(testData.context.broadcast(this))
+      case _ =>
+    }
     testData.mapPartitions { iter =>
-      val model = bcModel.value
+      val model = bcModel.get.value
```

Comment:
I believe we will still want to make a local reference to bcModel here so we don't end up shipping the entire object across the wire.